internal/tiltfile/tiltfile_state.go

```diff
@@ -1038,12 +1039,40 @@ func (s *tiltfileState) validateLiveUpdate(iTarget model.ImageTarget, g model.Ta
 			return fmt.Errorf("fall_back_on paths '%s' is not a child of any watched filepaths (%v)",
 				path, watchedPaths)
 		}
-
 	}
 
 	return nil
 }
 
+func (s *tiltfileState) maybeWarnRestartContainerDeprecation(manifests []model.Manifest) {
+	var needsWarn []model.ManifestName
+	for _, m := range manifests {
+		if needsRestartContainerDeprecationWarning(m) {
+			needsWarn = append(needsWarn, m.Name)
+		}
+	}
+
+	if len(needsWarn) > 0 {
+		s.logger.Warnf(restartContainerDeprecationWarning(needsWarn))
```

Comment:
this should be Warnf("%s", msg) so that Printf doesn't try to interpret anything in here as a format string